### PR TITLE
fix: DateRangerPicker with hh timeInputFormat

### PR DESCRIPTION
**Pull request by michaeldowseza**
Normalizes returned value when timeInput is of the format hh

### Description

Related links, issue #, if available: n/a

### How has this been tested?

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/date-range-picker/__tests__/date-range-picker-absolute.test.tsx [hunk 1]
```diff
@@ -241,6 +241,34 @@ describe('Date range picker', () => {
       });
     });
 
+    describe('time formats', () => {
+      test('correctly handles a time format of hh', () => {
```
Comment (just-boris) [inline src/date-range-picker/__tests__/date-range-picker-absolute.test.tsx:245]: How about adding a test for `hh:mm` too? To demonstrate consistency between modes

File: src/date-range-picker/__tests__/date-range-picker-absolute.test.tsx [hunk 2]
```diff
@@ -241,6 +241,34 @@ describe('Date range picker', () => {
...
+        const onChangeSpy = jest.fn();
+        const { wrapper } = renderDateRangePicker({
+          ...defaultProps,
+          timeInputFormat: 'hh',
+          onChange: event => onChangeSpy(event.detail),
+        });
+
+        act(() => wrapper.findTrigger().click());
```
Comment (just-boris) [inline src/date-range-picker/__tests__/date-range-picker-absolute.test.tsx:253]: Test utils already call `act` when needed. No need to do it one more time

File: src/date-range-picker/__tests__/date-range-picker-absolute.test.tsx (hunk 1 above)
Comment (michaeldowseza) [reply src/date-range-picker/__tests__/date-range-picker-absolute.test.tsx]: Added for all time formats 👍🏻

File: src/date-range-picker/__tests__/date-range-picker-absolute.test.tsx (hunk 2 above)
Comment (michaeldowseza) [reply src/date-range-picker/__tests__/date-range-picker-absolute.test.tsx]: Removed from my tests. The entire file wraps everything in act though. I can do a follow up PR to remove it from the rest as well.

Comment (just-boris) [reply src/date-range-picker/__tests__/date-range-picker-absolute.test.tsx]: Yes we have a lot of old code, pre-dating this change: https://github.com/cloudscape-design/test-utils/pull/25

Any cleanups would be nice, easy wins there

**Review by just-boris — APPROVED** [review]
Is approving a draft PR even possible?

**Pull request merged**